src/netbsd.rs

```diff
@@ -25,27 +41,33 @@ fn kern_arnd(buf: &mut [MaybeUninit<u8>]) -> libc::ssize_t {
 type GetRandomFn = unsafe extern "C" fn(*mut u8, libc::size_t, libc::c_uint) -> libc::ssize_t;
 
 // getrandom(2) was introduced in NetBSD 10.0
-static GETRANDOM: LazyPtr = LazyPtr::new();
+static GETRANDOM: AtomicPtr<c_void> = AtomicPtr::new(ptr::null_mut());
 
-fn dlsym_getrandom() -> *mut c_void {
+#[cold]
+fn init() -> *mut c_void {
     static NAME: &[u8] = b"getrandom\0";
     let name_ptr = NAME.as_ptr().cast::<libc::c_char>();
-    unsafe { libc::dlsym(libc::RTLD_DEFAULT, name_ptr) }
+    let ptr = match unsafe { libc::dlsym(libc::RTLD_DEFAULT, name_ptr) } {
+        ptr if !ptr.is_null() => ptr,
+        _ => polyfill_using_kern_arand as *mut libc::c_void,
+    };
```

Comment:
I think the following code will be a bit easier to read:
```rust
let mut ptr = unsafe { libc::dlsym(libc::RTLD_DEFAULT, name_ptr) };
// If getrandom(2) is not availalbe, use KERN_ARAND polyfill
if ptr.is_null() {
    ptr = polyfill_using_kern_arand as *mut libc::c_void;
}
```

Reply:
In other PRs, @josephlr asked me to write things the way I wrote it here, and to not write it the `let mut ... ` way.

Reply:
Ok. :) I don't have a strong preference here, so I will wait for him to confirm that he prefers this way.

Reply:
> In other PRs, @josephlr asked me to write things the way I wrote it here, and to not write it the `let mut ... ` way.

Looking at this, I think that was a bad call on my part (sorry). I agree w/ @newpavlov that the `if` is more readable here. @briansmith's point is a good one about code coverage, but we would ideally handle that via branch coverage (see https://github.com/rust-random/getrandom/issues/288#issuecomment-2177351590)